src/core/thread/address_resolver.cpp

```diff
@@ -485,6 +486,8 @@ otError AddressResolver::Resolve(const Ip6::Address &aEid, uint16_t &aRloc16)
         ExitNow();
     }
 
+    VerifyOrExit(aAllowAddressQuery, error = OT_ERROR_NOT_FOUND);
```

Comment:
This method itself, is rather tricky. I checked all paths to convince myself it is all ok. 👍 

May be adding a comment here (help  future developers/reader of codel). 
Something like this:

```
Note that if `aAllowAddressQuery` is `false` then the `entry` is definitely 
already in a list, i.e., we cannot not get here with `aAllowAddressQuery` being 
`false` and `entry` being a newly allocated one, due to the `VerifyOrExit` check 
that `aAllowAddressQuery` is `true` before allocating a new cache entry. 
```

Reply:
The comment you suggested is great.

I was trying to implement the instant `Resolve` without sending `Address Query` with minimal code increase. That's why I didn't add a new method for this purpose.